core/src/task.rs

```diff
@@ -533,48 +537,151 @@ macro_rules! vdaf_dispatch {
     };
 }
 
-/// An authentication (bearer) token used by aggregators for aggregator-to-aggregator and
-/// collector-to-aggregator authentication.
-#[derive(Clone)]
-pub struct AuthenticationToken(Vec<u8>);
+/// Different modes of authentication supported by Janus for either sending requests (e.g., leader
+/// to helper) or receiving them (e.g., collector to leader).
+#[derive(Clone, Derivative, Serialize, Deserialize)]
+#[derivative(Debug)]
+#[serde(tag = "type", content = "token")]
+#[non_exhaustive]
+pub enum AuthenticationToken {
+    /// A bearer token. The value is an opaque byte string. Its Base64 encoding is inserted into
+    /// HTTP requests as specified in [RFC 6750 section 2.1][1]. The token is not necessarily an
+    /// OAuth token.
+    ///
+    /// [1]: https://datatracker.ietf.org/doc/html/rfc6750#section-2.1
+    Bearer(
+        #[derivative(Debug = "ignore")]
+        #[serde(
+            serialize_with = "as_url_safe_unpadded_base64",
+            deserialize_with = "from_url_safe_unpadded_base64"
```

Comment:
Since this is base64-encoded on the wire in headers, what do you think about changing the YAML/JSON serialization to use base64 as well?

Reply:
Yeah, I think using plain base64 makes more sense in a config file, as URL safety doesn't matter there. It's inconsistent with how we encode DAP-Auth-Tokens, but I can live with that wart.